marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs

```diff
@@ -1,13 +1,25 @@
 const lib = require('/data-hub/5/impl/hub-es.sjs');
 
-var entityTypeId;
-var propertyPath;
-var referenceType;
-var pattern;
-var limit;
+var facetValuesSearchQuery;
 
+let queryObj = JSON.parse(facetValuesSearchQuery);
+let entityTypeId = queryObj.facetInfo.entityTypeId;
```

Comment:
Can we handle possible NPE's?

Reply:
Is facetInfo guaranteed to be non-null? I think including the JSON schema for a search query in the new ./specs folder would be a good step forward to helping anyone know the answer to this.